### DISCUSSION TITLE
Allow overriding the default JSON encoder library

### DISCUSSION THREAD

**Discussion by radupotop** [encode/httpx#2626] [Ideas]
I'm in the process of migrating from the build-in `json` encoder library to the much faster `Orjson` library, in a rather large project. One of the approaches I've been looking at is to monkey-patch the `requests` library, so that I don't have to rewrite all the handlers.

This is easier said than done unfortunately. So I'm now looking at httpx, but unfortunately I would still have to monkey-patch the `_main` and the `_models` modules.

It would be great if there way a way to specify custom `dumps` and `loads` handlers for JSON.

For example Pydantic supports this using the json_loads and json_dumps properties in its Config: https://docs.pydantic.dev/usage/exporting_models/#custom-json-deserialisation

**Comment by Kludex** [comment]
https://github.com/encode/httpx/issues/717
